modules/standard/IO.chpl

```diff
@@ -2924,30 +2924,23 @@ inline operator :(x: ioLiteral, type t:string) {
   return x.val;
 }
 
-/*
-
-Represents a value with a particular bit length that we want to read or write.
-The I/O will always be done in binary mode.
-
-*/
-record ioBits {
-  /* The bottom ``nbits`` of v will be read or written */
-  var v:uint(64);
-  /* How many of the low-order bits of ``v`` should we read or write? */
-  var nbits:int(8);
-  pragma "no doc"
-  proc writeThis(f) throws {
-    // Normally this is handled explicitly in read/write.
-    f.write(v);
-  }
+pragma "no doc"
+record _internalIoBits {
+  /* The bottom ``numBits`` of x will be read or written */
+  var x:uint(64);
+  /* How many of the low-order bits of ``x`` should we read or write? */
+  var numBits:int(8);
 
-  pragma "no doc"
-  proc encodeTo(f) throws {
-    f._writeOne(f.kind, this, here);
-  }
+  // keep the old names for compatibilty with old ioBits
+  proc v: x.type {return x;}
+  proc nbits:numBits.type {return numBits;}
 }
 
+deprecated "ioBits type is deprecated - please use :proc:`fileReader.readBits` and :proc:`fileWriter.writeBits` instead"
+type ioBits = _internalIoBits;
+
 pragma "no doc"
+pragma "compiler generated"
```

Comment:
Why does this have this pragma added? I would not expect this pragma to be needed here. Should this function be using `_internalIoBits` in the function formal's type?

Reply:
Added pragma to silence deprecation on internal use, I agree it makes more sense to just change the function to use `_internalIoBits`